src/views/DepositWithdraw/components/EnterAmount/index.tsx

```diff
@@ -21,37 +23,38 @@ import {
 } from './components';
 import { H2, P2 } from 'components/Typography';
 
-import { TOKEN_GEYSER_ADDRESS } from 'constants/index';
-
 import { BalanceContext } from 'contexts/Balance';
+import { DepositWithdrawalContext } from 'contexts/DepositWithdrawal';
 import { Web3Context } from 'contexts/Web3';
 import { API as OnboardAPI } from 'libs/types';
 
-import { parseInputValue, stake } from 'utils';
-
 import useApproval from 'hooks/useApproval';
 
 interface IEnterAmount {
+  actionType: 'deposit' | 'withdraw';
   farmSelected: boolean;
   readyToTransact: (onboard: OnboardAPI | null, provider: Provider | null) => Promise<boolean>;
   setFarmSelected: React.Dispatch<React.SetStateAction<boolean>>;
 }
 
-export const EnterAmount: React.FC<IEnterAmount> = ({ farmSelected, readyToTransact, setFarmSelected }) => {
-  const { wpoktBalance } = React.useContext(BalanceContext);
-  const [isDisabled, setIsDisabled] = React.useState<boolean>(true);
-  const [wpoktInputValue, setWpoktInputValue] = React.useState<string>('');
+export const EnterAmount: React.FC<IEnterAmount> = ({ actionType, farmSelected, readyToTransact, setFarmSelected }) => {
+  const { inputValue, onChangeInput, onSelectModal } = React.useContext(DepositWithdrawalContext);
   const { address, onboard, provider, signer } = React.useContext(Web3Context);
+  const { wpoktBalance } = React.useContext(BalanceContext);
+
   const { isApproved, isApproving, onApprove } = useApproval();
 
+  const [isDisabled, setIsDisabled] = React.useState<boolean>(true);
+
   React.useEffect(() => {
-    if (wpoktInputValue === '' || wpoktInputValue === '0') {
+    if (inputValue === '' || inputValue === '0') {
       setIsDisabled(true);
+      setFarmSelected(false);
     } else {
       setIsDisabled(false);
       setFarmSelected(true);
     }
-  }, [address, setFarmSelected, wpoktInputValue]);
+  }, [address, setFarmSelected, inputValue]);
 
   // This is a placeholder, which will eventually launch the transaction status modal
```

Comment:
We can probably remove this now :)

Reply:
😅 71975a4375568e3aabea19bccf7b2865a5342264